lsls/lsls.c

```diff
@@ -1,18 +1,55 @@
 #include <stdio.h>
 #include <dirent.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <string.h>
 
 /**
  * Main
  */
 int main(int argc, char **argv)
 {
-  // Parse command line
+  // Define dirname based on arg count
+  const char *dirname;
+  if (argc == 1)
+  {
+    dirname = ".";
+  }
+  else
+  {
+    dirname = argv[1];
+  }
 
   // Open directory
+  DIR *dir;
+  struct dirent *entry;
 
-  // Repeatly read and print entries
+  // Error handling
+  if ((dir = opendir(dirname)) == NULL)
+  {
+    fprintf(stderr, "Cannot open %s\n", dirname);
+    exit(1);
+  }
 
+  // Repeatly read and print entries
+  while ((entry = readdir(dir)) != NULL)
+  {
+    struct stat buf;
+    char fullpath[1024] = "";
+    strcat(fullpath, dirname);
```

Comment:
This is a very creative way to implement this. You should research `snprintf()` for an alternative to calling multiple strcat calls in order to fill a buffer.